docs/DeveloperGuide.md

```diff
@@ -1,33 +1,373 @@
+
+The following components processes their respective actions (parsing and execution) and handles their own
+storage component: 
+* **Bookmark**
+* **Timetable**
+* **Academic**
+* **Flashcard**
+
+**General commands** are commands that takes priority in execution no matter which mode the software is currently
+in.
+
+**Modes of the program**
+
+Study It has 5 **modes** that the user can navigate between:
+* **Main menu**: Default mode when booting the software, there are no functionalities tied to this mode
+* **Bookmark**
+* **Timetable**
+* **Academic**
+* **Flashcard**
+
+Each of these modes has their own unique commands. The user must switch to that mode to perform the
+mode's functionalities.
+
+User can switch between these modes at any point when using the software.
+
+**How the architecture components interact with each other**
+
+![Sequence Overview](Images/ArchitectureDiagrams/overviewsequence.png)
+
+As shown in the sequence diagram above, the user will interact with Study It via the User Interface (UI).
+The commands received will then be parsed to identify the type of command. If the command is one of the 
+general commands, it'll be processed and provide output. Otherwise, the command will be relayed
+to different components to be processed based on the software's current mode. 
+
+### **Main Component**
+This section will give an overview of the main component of Study It
+
+![Main Component Diagram](Images/MainComponentDiagrams/maincomponent.png)
+
+The main class being run for the program is contained under `StudyIt` class. 
+As there are multiple modes of our app (main menu/bookmark/timetable/academic/flashcard), 
+the Mode enumeration and a private static currentMode of Mode type is introduced to monitor the current 
+operating mode of StudyIt. This is to allow the program to respond to different commands under different modes.
+
+Each command input by the user will be processed by CommandParser class to determine the command type. 
+This information, along with the original command string will be passed into Command class and executed under 
+executeCommand(). If it is one of the general commands, it will be processed by that class. 
+If it isn’t, the Command class will detect the current mode of the program and run that command string under that 
+specific mode, distributed under handleNonGeneralCommand() method. If it still doesn’t parse successfully, 
+it will be considered an invalid command and an error message will be printed out.
+
+The classes Ui, ErrorMessage, HelpMessage and MainMenu help handle the printing of various user interfaces of the 
+program. ErrorMessage, HelpMessage and MainMenu are subclasses of Ui to make use of the methods in Ui.
+
+StudyIt class will also initialize various instances of classes such as TimeTableRun, FlashCardRun, 
+ArrayList<BookmarkCategory>, ArrayList<Grade>, ArrayList<Person> and pass it to Command class to perform 
+each mode’s functionality.
```

Comment:
Maybe surround all classes like Ui, ErrorMessage and so on with ` ` instead of just StudyIt.